mlir-sycl/lib/Conversion/SYCLToSPIRV/SYCLToSPIRV.cpp

```diff
@@ -199,9 +199,11 @@ class GridOpPatternNoIndex : public GridOpPattern<OpTy> {
 void rewrite1D(Operation *op, spirv::BuiltIn builtin,
                const SPIRVTypeConverter &typeConverter,
                ConversionPatternRewriter &rewriter) {
-  rewriter.replaceOp(op,
-                     spirv::getBuiltinVariableValue(
-                         op, builtin, typeConverter.getIndexType(), rewriter));
+  const auto res = spirv::getBuiltinVariableValue(
+      op, builtin, typeConverter.getIndexType(), rewriter);
+  rewriter.replaceOp(op, convertScalarToDtype(rewriter, op->getLoc(), res,
+                                              op->getResultTypes()[0],
```

Comment:
Should we convert to the converted type using `typeConverter()->convertType()`?

Reply:
Well, that should be the very same type everytime, as the result is `i32`, but I guess it doesn't hurt to be extra cautious. Added.